web/modules/custom/joinup_licence/src/PathProcessor/SpdxIdsPathProcessor.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_licence\PathProcessor;
+
+use Drupal\Core\PathProcessor\InboundPathProcessorInterface;
+use Drupal\Core\PathProcessor\OutboundPathProcessorInterface;
+use Drupal\Core\Render\BubbleableMetadata;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Defines a path processor to rewrite the licence comparer URL.
+ *
+ * As the route system does not allow arbitrary amount of parameters convert
+ * the file path to a query parameter on the request.
+ */
+class SpdxIdsPathProcessor implements InboundPathProcessorInterface, OutboundPathProcessorInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processInbound($path, Request $request): string {
+    if (strpos($path, '/licence/compare/') === 0 && !$request->query->has('spdx_ids')) {
```

Comment:
Continuing from the comment with the `+` character in the SPDX licence, I have two edge cases here:
* `/licence/compare/AFL-3.0/GPL-2.0+` ends here in the `::processInbound` method as `/licence/compare/AFL-3.0/GPL-2.0 `.
* More dangerous is the case where the `+` symbol is not in the last position. I am trying to access the URL `/licence/compare/AFL-3.0+/GPL-2.0`. For some reason, the `$path` variable here contains the value `/licence/compare/AFL-3.0 /piwik.js`. The page is still ending up in a not found exception, but that might mean that the URL processor is vulnerable.

Reply:
The IDs in route path is a mess and doesn't work. I switched to query string.